tests/js/client/shell/api/cursor.js

```diff
@@ -1212,9 +1212,268 @@ function dealing_with_cursorsSuite_fetching_floating_point_valuesSuite () {
+
+function dealing_with_cursorsSuite_retriable_request_last_batch() {
+  let cn = "users";
+  return {
+    setUpAll: function () {
+      db._drop(cn);
+      let c = db._create(cn);
+
+      let docs = [];
+      for (let i = 0; i < 2001; i++) {
+        docs.push({"_key": `test${i}`});
+      }
+      c.insert(docs);
+    },
+
+    tearDownAll: function () {
+      internal.debugClearFailAt();
+      db._drop(cn);
+    },
+
+    test_cursor_non_stream_request_retriable: function () {
+      let cmd = api;
+      let body = {"query": `FOR u IN ${cn} RETURN u`, "options": {"stream": false, "retriable": true}};
+      let doc = arango.POST_RAW(cmd, body);
+
+      assertEqual(doc.code, 201);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertFalse(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 201);
+      assertEqual(typeof doc.parsedBody['id'], "string");
+      assertMatch(reId, doc.parsedBody['id']);
+      assertTrue(doc.parsedBody['hasMore']);
+      assertEqual(doc.parsedBody['result'].length, 1000);
+      assertFalse(doc.parsedBody['cached']);
+
+      let cursorId = doc.parsedBody['id'];
+      assertEqual(doc.parsedBody['nextBatchId'], 2);
+      let nextBatchId = doc.parsedBody['nextBatchId'];
+
+      internal.debugSetFailAt("MakeConnectionErrorForRetry");
+
+      cmd = api + `/${cursorId}`;
+      doc = arango.POST_RAW(cmd, "");
+
+      assertEqual(doc.code, 500);
+      assertTrue(doc.error);
+
+      internal.debugClearFailAt();
+
+      cmd = api + `/${cursorId}/${nextBatchId}`;
+      doc = arango.POST_RAW(cmd, "");
+
+      assertEqual(doc.code, 200);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertFalse(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 200);
+      assertTrue(doc.parsedBody['hasMore']);
+      assertEqual(doc.parsedBody['result'].length, 1000);
+      assertFalse(doc.parsedBody['cached']);
+
+      assertEqual(doc.parsedBody['nextBatchId'], 3);
+      nextBatchId = doc.parsedBody['nextBatchId'];
+      cmd = api + `/${cursorId}/${nextBatchId}`;
+      doc = arango.POST_RAW(cmd, "");
+
+      assertEqual(doc.code, 404);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertTrue(doc.parsedBody.error);
+      assertEqual(doc.parsedBody.errorNum, 404);
+      assertEqual(doc.parsedBody.errorMessage, "batch id not found");
+      assertEqual(doc.parsedBody['code'], 404);
+
+      cmd = api + `/${cursorId}`;
+      doc = arango.POST_RAW(cmd, "");
+
+      assertEqual(doc.code, 200);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertFalse(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 200);
+      assertFalse(doc.parsedBody['hasMore']);
+      assertEqual(doc.parsedBody['id'], undefined);
+      assertEqual(doc.parsedBody['result'].length, 1);
+      assertFalse(doc.parsedBody['cached']);
```

Comment:
Are we expected a `nextBatchId` in the response here`? If not, we should verify that the response does not contain one.